### PR TITLE
`Box` component `0` value fix

### LINKED ISSUE #16976: 0 doesn't work for bounding box values for Box component
### Description
The value `0` doesn't work for box because there is a boolean check before running the `generateClassnames` function.

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## Explanation

**What is the current state of things and why does it need to change?**
Currently `0` value option does not work on the `Box` component margin or padding props. The CSS styles exists but the classNames are not being generated. This was introduced during the responsive props update https://github.com/MetaMask/metamask-extension/pull/15106 and completely my fault 🤦 

**What is the solution your changes offer and how does it work?**
- The `Box` component logic has been updated to resolve the `0` value issue and now generates the correct class names
- Tests have been update to have a more comprehensive coverage for margin, padding and gap.
- Responsive story has been updated to show more responsive prop examples.
- Snapshots updated with missing margin/padding 0 class names

This update removes the `&&` operator during the `generateClassNames` call for each prop and instead does a value check inside of `generateClassNames`. This means `generateClassNames` is called regardless of it's prop value. I've run some basic performance tests and couldn't see a huge change but open to suggestions on this. `0.4ms` vs `0.466ms` after the update run in an incognito window using the react dev tools profiler have documented results below.

* Fixes #16976 

### Before

https://user-images.githubusercontent.com/8112138/208551691-7e030506-39cc-4ec8-b06a-aa3fb0436311.mov

https://user-images.githubusercontent.com/8112138/208551721-2031caaf-e6ff-4029-a164-fe42b1134b7e.mov

#### Profiler testing performance
- Render 1: `0.4ms`
- Render 2: `0.3ms`
- Render 3: `0.5ms`

Average of `0.4ms`

Screenshot 2022-12-19 at 4 25 03 PM
Screenshot 2022-12-19 at 4 25 40 PM
Screenshot 2022-12-19 at 4 26 16 PM

### After

https://user-images.githubusercontent.com/8112138/208551746-5643e43a-f459-4f1e-bdf4-dcdd93380e40.mov

https://user-images.githubusercontent.com/8112138/208551766-f24ec980-4f3a-4ed5-ae61-5017cc714c72.mov

#### Profiler testing performance
- Render 1: `0.5ms`
- Render 2: `0.7ms`
- Render 3: `0.2ms`

Average of `0.466ms`

Screenshot 2022-12-19 at 4 22 39 PM
Screenshot 2022-12-19 at 4 23 22 PM
Screenshot 2022-12-19 at 4 24 06 PM

## Manual Testing Steps

- Go to the latest storybook build of this PR
- Search `Box` in the search panel
- Update any of the bounding box controls (margin, marginTop, padding, paddingTop, etc) to use `0` value
- In developer tools check that the className and CSS have been added correctly

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by garrettbear — DISMISSED** [review]

**Review by NidhiKJha — DISMISSED** [review]

**Review by garrettbear — DISMISSED** [review]
Still LGTM! Just tested and working well.

File: test/e2e/tests/add-account.spec.js [hunk 1]
```diff
@@ -33,7 +33,10 @@ describe('Add account', function () {
       },
       async ({ driver }) => {
         await driver.navigate();
-        await driver.fill('#password', 'correct horse battery staple');
+        await driver.pasteIntoField(
+          '#password',
+          'correct horse battery staple',
+        );
```
Comment (georgewrmarshall) [inline test/e2e/tests/add-account.spec.js:39]: Attempting to update test to prevent timeout

**Comment by georgewrmarshall** [comment]
Can't seem to get e2e tests to pass 🤔

File: ui/components/ui/box/box.stories.tsx [hunk 2]
```diff
@@ -652,6 +653,51 @@ export const ResponsiveProps = () => {
...
+        This box is only visible on screens 768px and above
+      </Box>
+      <Box
+        padding={4}
+        display={[DISPLAY.NONE, DISPLAY.BLOCK]}
+        borderColor={Color.borderDefault}
+      >
+        This box is only visible on screens 576px and above
+      </Box>
     </>
```
Comment (georgewrmarshall) [inline ui/components/ui/box/box.stories.tsx:701]: This could probably go in another PR but responsive story for display


https://user-images.githubusercontent.com/8112138/227387800-991a16bd-a660-4f0b-bc99-968edf6202c5.mov

**Comment by brad-decker** [comment]
@georgewrmarshall if you wanna rebase this, and then tag me on it i'll take a look at the e2e.

**Comment by georgewrmarshall** [comment]
@brad-decker I would greatly appreciate that 🙏 Rebased and snapshots up to date!

**Comment by georgewrmarshall** [comment]
RIP 🪦 on your bones a new typescript box update will be built upon. Moved to https://github.com/MetaMask/metamask-extension/pull/19363
